File: src/go/rsyncer/lib/commandverify.go
package lib

import (
	"fmt"
	"os/exec"
)

func Commandverify(cmd string) bool {
	check := exec.Command("bash", "-c", fmt.Sprintf("type %s",cmd))
	_, err := check.Output();

	if(err == nil) {
		return true
	} else {
		return false
	}
}

File: src/dart/rsyncer/lib/commandverify.dart
import 'dart:io' show Process;

Future<bool> commandverify(String command) async {
    final check = await Process.run("bash", ["-c", "type $command"]);
    if(check.exitCode == 0) {
      return true;
    } else {
      return false;
    }
}